src/util/job-name-validation.tsx

```diff
@@ -64,3 +65,63 @@ export function NameError(name: string, trans: TranslationBundle): string {
     'Name must contain only letters, numbers, spaces, periods, hyphens, and underscores'
   );
 }
+
+export function MaxRetryAttemptsError(maxRetryAttempts: string, trans: TranslationBundle): string {
+  // Check for blank
+  if (maxRetryAttempts === '') {
+    return trans.__('You must specify max retry attempts');
+  }
+
+  if (!validIntegerRregex.test(maxRetryAttempts)) {
+    return trans.__(
+      'Max retry attempts must be an integer'
+    );
+  }
+ const integerValue = +maxRetryAttempts
+  // Check for length.
+  if (integerValue < 1 || integerValue > 30) {
```

Comment:
Use symbolic constants to indicate the minimum and maximum acceptable values.

Why isn't 0 an acceptable value? It would fit if I wanted a job to be tried only once, and never retried if it fails.